firmware/library/L3_HAL/buzzer.hpp

```diff
@@ -0,0 +1,43 @@
+// Usage:
+// initialize buzzer with pwm pin number
+// BuzzerInterface * buzzer = new Buzzer(1) ;
```

Comment:
Why does your example require the usage of new? The internal libraries should not dynamic memory, nor should their examples, unless the example is specifically made to showcase dynamic memory. You can change this to:

    Buzzer buzzer(1);
    buzzer.Beep(500, 0.9);